test/testutil/driver.c

```diff
@@ -280,20 +280,20 @@ void set_test_title(const char *title)
     test_title = title == NULL ? NULL : strdup(title);
 }
 
-PRINTF_FORMAT(2, 3) static void test_verdict(int pass, const char *extra, ...)
+PRINTF_FORMAT(2, 3) static void test_verdict(int verdict,
+                                             const char *description, ...)
 {
     va_list ap;
 
     test_flush_stdout();
     test_flush_stderr();
 
-    test_printf_stdout("%*s%s", level, "", pass ? "ok" : "not ok");
-    if (extra != NULL) {
-        test_printf_stdout(" ");
-        va_start(ap, extra);
-        test_vprintf_stdout(extra, ap);
-        va_end(ap);
-    }
+    test_printf_stdout("%*s%s ", level, "", verdict != 0 ? "ok" : "not ok");
+    va_start(ap, description);
+    test_vprintf_stdout(description, ap);
+    va_end(ap);
+    if (verdict == TEST_SKIP_CODE)
+        test_printf_stdout(" # skip");
```

Comment:
When it's a skip, the description and vararg is the skip explanation, isn't it?  In that case, the output should be this:

```
ok 13 # skip Whatever
```

What you will get right now is this:

```
ok 13 - Whatever # skip
```

Trouble is that some of this is formatted in `run_tests`...  however, that can be fixed by having `test_verdict` take the test number as a separate argument, so that `test_verdict` can have better control over the line to output.

Reply:
Ref: https://testanything.org/tap-specification.html (look for "Skipping tests")

Reply:
The test output isn't giving a reason for the skip which seems to be the "ok 13 # skip ..." case.  The format is:

1. ok/not ok (required)
2. number (recommended)
3. description (recommended)
4. directive (only when necessary)

The directive can include trailing text, but I don't see how to pass the skip reason back to be output in such a fashion.  Instead, it is output on the previous line with a SKIP annotation.

Reply:
i.e. I think the current output matches the definition for a skipped test.  The only ommission being a reason for the skip which isn't available at that point in the code.

Reply:
I'd prefer `skipped: ` instead of `skip `,

```
ok 13 # skipped: whatever
```
because this fits better to the normal skipped messages.

```
...
../test/recipes/05-test_idea.t ..................... ok
../test/recipes/05-test_md2.t ...................... skipped: md2 is not supported by this OpenSSL build
../test/recipes/05-test_mdc2.t ..................... ok
../test/recipes/05-test_rand.t ..................... ok
../test/recipes/05-test_rc2.t ...................... ok
../test/recipes/05-test_rc4.t ...................... ok
../test/recipes/05-test_rc5.t ...................... skipped: rc5 is not supported by this OpenSSL build
../test/recipes/06-test-rdrand.t ................... ok
...
```